### ISSUE TITLE
Add support for non-EVM chains

### ISSUE THREAD

**Issue by 0xmichalis** [rolod0x/rolod0x#419]
Currently it seems that only EVM addresses are supported. It'd be great to add support for any non-EVM chain too like Bitcoin, Solana, Tezos, etc.

**Comment by aspiers** [comment]
Weird, thought I already had an issue for this, but yeah it's an obvious gap.  Not a personal priority right now for me but PRs definitely welcome.

**Comment by joeyfigaro** [comment]
@aspiers I'm currently working on this–just a heads up. 💪🏻

**Comment by aspiers** [comment]
@joeyfigaro Awesome!

**Comment by joeyfigaro** [comment]
> @joeyfigaro Awesome!

would you prefer to just expand on the existing regex patterns to handle identifying address strings?

**Comment by aspiers** [comment]
Hrm, depends on the third party deps I guess.  What did you have in mind?  In general I'm a fan of not reinventing wheels so if there are good libraries fit for purpose then maybe that's best, but the devil's often in the details.

**Comment by joeyfigaro** [comment]
> Hrm, depends on the third party deps I guess. What did you have in mind? In general I'm a fan of not reinventing wheels so if there are good libraries fit for purpose then maybe that's best, but the devil's often in the details.

I've found a few one off libraries for validating btc addresses, but they have all come with a lot more than is needed–things like transaction signing and wallet creation, which we obviously don't need for this effort. 

I'll do some digging / info collecting and circle back.

**Comment by aspiers** [comment]
It's worth mentioning that I use ethers.js for validating EVM addresses. That said, regexps are still needed for detection and replacement. Probably the way to go is to copy how I've done it for EVM.

**Comment by 0xmichalis** [comment]
Probably worth to break this issue down to per chain.

**Comment by aspiers** [comment]
Agreed.  The only other chain I personally care about is Bitcoin.  Since we already have #490 for Solana, let's repurpose this one for Bitcoin and open others as needed.

**Comment by aspiers** [comment]
@joeyfigaro If you are still interested in adding Bitcoin support you should definitely keep an eye on #490.

**Comment by joeyfigaro** [comment]
Perfect @aspiers–thank you! Circling back to this after a much needed hiatus.
